docs/docs/reference/dependent-function-types.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: doc-page
+title: "Dependent Function Types"
+---
+
+A dependent function type describes functions where the result type may depend
+on the function's parameter values. Example:
+
+    class Keyed { type Key; key: Key }
+
+    def extractKey(k: Keyed): k.Key = k.key          // a dependent method
+    val extractor: (k: Keyed) => k.Key = extractKey  // a dependent function value
+
+Scala already has _dependent methods_, i.e. methods where the result
+type refers to some of the parameters of the method. Method
+`extractKey` is an example. Its result type, `k.key` refers its
+parameter `k` (we also say, `k.Key` _depends_ on `k`). But so far it
+was not possible to turn such methods into function values, so thay
+they can be passed as parameters to other functions, or returned as
+results. Dependent methods could not be turned into functions simply
+because there was no type that could describe them.
+
+In Dotty this is now made possible. The type of the `extractor` value above is
+
+    (k: Keyed) => k.Key
+
+This type describes function values that take an argument `x` of type
```

Comment:
It's `k`, not `x` (same on the next line)

Reply:
No, `k` is the parameter. I wanted to use a different name for the argument.